### DISCUSSION TITLE
go build in formula -o overwritten

### DISCUSSION THREAD

**Discussion by cdenneen** [Homebrew/discussions#5367] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
 brew config
HOMEBREW_VERSION: 4.3.0-18-gf4a9869
ORIGIN: https://github.com/Homebrew/brew
HEAD: f4a9869d27f5f3ece082e74da38a5a06a67a2637
Last commit: 4 hours ago
Core tap HEAD: 03e0d6738512f967c3aa6312b7e6bf36bf2b7ef5
Core tap last commit: 3 hours ago
Core tap JSON: 14 May 20:58 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: localhost:10.0
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 2
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: dual-core 64-bit skylake
Clang: N/A
Git: 2.45.0 => /home/linuxbrew/.linuxbrew/bin/git
Curl: 8.3.0 => /bin/curl
Kernel: Linux 5.10.215-203.850.amzn2.x86_64 x86_64 GNU/Linux
OS: Amazon Linux 2
Host glibc: 2.26
/usr/bin/gcc: 7.3.1
/usr/bin/ruby: N/A
glibc: 2.35_1
gcc@11: N/A
gcc: 14.1.0
xorg: N/A
```

### Output of `brew doctor`

```shell
 brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  packer
  terraform
```

### Description of issue

I am currently using
`system "go", "build", *std_go_args(ldflags:), "./cmd/main.go"` because when using `system "go", "build", "-o hack/switch/switcher", *std_go_args(ldflags:), "./cmd/main.go"` it never renders the binary in hack/switch/switcher. I was trying to use this output in order to do the `bin.install` and avoid versioned `switcher@0.9.1` binaries being built.

**Comment by Bo98** [comment]
`std_go_args` has an optional `output` argument you can pass: `std_go_args(output: bin/"switcher")`

**Reply by cdenneen** [reply]
Is there a way to write this so it would do the linking properly?
For example if I'm using the latest formula it would be bin/"switcher"... if I'm using SwitcherAT091 then it would be bin/"switcher@0.9.1", if I'm using SwitcherAT072 it would be bin/"switcher@0.7.2" for example? I think the ability to run different versions by calling different binaries is useful possibly but would need a way for things like `generate_completions_from_executable(bin/"switcher", "completion", base_name: "switcher")` to work when you are installing a version of a formula.
Or maybe when you install a version of a formula it creates a link from the current version binary that gets added like `switcher@0.9.1` to `switcher` with link... so it would update the brew unlink/link and then the `generate_completions_from_executable` would update completions based on the current version installed. 
I could then unlink/link between versions but if I'm installing a new version most likely I'm going to want to create the link to that new version  installed.

**Reply by cdenneen** [reply]
```
  def install
    rm_rf ".brew_home"
    ldflags = %W[
      -w
      -X github.com/danielfoehrkn/kubeswitch/cmd/switcher.version=#{version}
      -X github.com/danielfoehrkn/kubeswitch/cmd/switcher.buildDate=#{time.iso8601}
    ]
    std_go_args(output: bin/"switcher")

    ENV.prepend_path "PATH", Formula["coreutils"].opt_libexec/"gnubin" # needs GNU date
    system "go", "build", *std_go_args(ldflags:), "./cmd/main.go"
    bin.install "switcher"

    generate_completions_from_executable(bin/"switcher", "completion", base_name: "switcher")
  end
```
I've also tried:
```
    system "go", "build", *std_go_args(ldflags:, output: bin/"switcher"), "./cmd/main.go"
```

All result in:
```
Error: An exception occurred within a child process:
  Errno::ENOENT: No such file or directory - switcher
```

**Reply by carlocab** [reply]
That comes from `bin.install "switcher"`. Get rid of it; you don't need it.
